go/teams/chain.go

```diff
@@ -1518,18 +1530,49 @@ func (t *TeamSigChainPlayer) addInnerLink(
 							if err != nil {
 								return err
 							}
-							if _, err := prevState.GetLatestUVWithUID(uv.Uid); err != nil {
-								return err
+							addedUIDs[uv.Uid] = true
+							_, err = prevState.GetLatestUVWithUID(uv.Uid)
+							if err == nil {
+								// Found crypto member in previous
+								// state, we are good!
+								continue
 							}
-						} else {
-							return fmt.Errorf("invalid invite type in implicit team: %v", cat)
+							_, _, found := prevState.FindActiveKeybaseInvite(uv.Uid)
+							if found {
+								// Found PUKless member in previous
+								// state, still fine!
+								continue
+							}
+							// Neither crypto member nor PUKless member
+							// found, we can't allow this addition.
+							return fmt.Errorf("Not found previous version of user %s", uv.Uid)
 						}
+						return fmt.Errorf("invalid invite type in implicit team: %v", cat)
+					}
+				}
+
+				var cancelledUVs []keybase1.UserVersion
+				for _, inviteID := range cancelations {
+					invite, found := prevState.FindActiveInviteByID(inviteID)
+					if !found {
+						return fmt.Errorf("encountered cancellation of invite that doesn't exist")
```

Comment:
It might be good to allow canceling an invite that doesn't exist because it's harmless. Also it might be canceling an obsolete invite

Reply:
good catch! thanks!